core/domain/exp_services.py

```diff
@@ -1754,15 +1784,9 @@ def revert_exploration(
             'which is too old. Please reload the page and try again.'
             % (exploration_model.version, current_version))
 
-    # Validate the previous version of the exploration before committing the
-    # change.
+    # Assume revert_to_version passes all backend validation checks.
     exploration = exp_fetchers.get_exploration_by_id(
         exploration_id, version=revert_to_version)
-    exploration_rights = rights_manager.get_exploration_rights(exploration.id)
-    if exploration_rights.status != rights_domain.ACTIVITY_STATUS_PRIVATE:
-        exploration.validate(strict=True)
-    else:
-        exploration.validate()
```

Comment:
Why we remove it from here? Don't we still want to keep the check if someone forgets to first call the verification handler?

Reply:
Right now, there is only a single code path that calls `revert_exploration` (and it starts when a lesson creator clicks "Revert" in the history tab). I don't really see any future code paths being added, so seems like a safe bet that the validation doesn't need to be called twice. What do you think?

Reply:
I would still keep it if the validation is fairly quick (which I think is) or you can add an argument that would skip the check but it would not skip it by default.

Reply:
Alright! Added it back.